eth_test_parser/src/main.rs

```diff
@@ -31,9 +39,33 @@ fn run(ProgArgs { no_fetch }: ProgArgs) -> Result<()> {
         prepare_output_dir()?;
     }
 
-    // TODO: Use deserialized test structs to construct plonky2 generation inputs.
-    for (test_dir_entry, test_body) in get_deserialized_test_bodies()? {
-        println!("deserialized test {:?}: {:?}", test_dir_entry, test_body);
+    println!("Converting test json to plonky2 generation inputs");
+    let generation_inputs_handle =
+        get_deserialized_test_bodies()?.map(|(test_dir_entry, test_body)| {
+            thread::spawn(move || {
+                (
+                    test_dir_entry,
+                    serde_cbor::to_vec(&ParsedTest {
```

Comment:
Haha if we want to optimize further, we might want to look at using green threads.